### ISSUE TITLE
Some problem with iterator using BAI index

### ISSUE THREAD

**Issue by HatakeKakaxi** [samtools/htslib#1305]
I tried two ways to read the BAM file, one iterating from the start of the file, while another using index. When using index, I tried to iterate using _hts_itr_query_ method in htslib. Eveytime I got a hts_itr_t struct which can be used to iterate in a single chromosome. The code is like this:
```
    hts_idx_t *idx = NULL;
    char *fn_idx_in = 0;
    // If index filename has not been specfied, look in BAM folder
    if (fn_idx_in != NULL) { 
        idx = sam_index_load2(hts, in, fn_idx_in); 
    } else {
        idx = hts_idx_load3(in, NULL, HTS_FMT_BAI, HTS_IDX_SAVE_REMOTE);
    }

   for (int i = 0; i < idx->n; ++i) {
        hts_itr_t *iter = hts_itr_query(idx, i, 0, HTS_POS_MAX ,sam_readrec);
        ......    
    }
```
However, I found that the number of alignments in this way is less than the number I got by iterating from the start of of the file.  For example, I got 12823475 alignments using index, while getting 13217485 alignments in another way. So why does this happen? Can I iterate all the reads using BAI index ?

**Comment by whitwham** [comment]
I think this is the same problem as #1279 which has been fixed in #1281.  If you can use the latest development version of HTSlib then it should work.  A new release is also being prepared but it is not quite ready yet.

**Comment by HatakeKakaxi** [comment]
> I think this is the same problem as #1279 which has been fixed in #1281. If you can use the latest development version of HTSlib then it should work. A new release is also being prepared but it is not quite ready yet.

I got it. My problem has been solved. Thank you.
